cmd/cmp/main.go

```diff
@@ -52,23 +52,34 @@ func Run(_ *cobra.Command, args []string) {
 		}
 		if match := fileIdRegex.FindStringSubmatch(entry.Name()); len(match) > 0 {
 			fileId := match[1]
-			probeSideFileCount++
 			buildSideFilePath, exists := fileIdMap[fileId]
 			if !exists {
 				continue
 			}
+			probeSideFileCount++
 			probeSideFilePath := filepath.Join(probeSidePath, entry.Name())
-			buildSideString, probeSideString := readFileIntoString(buildSideFilePath), readFileIntoString(probeSideFilePath)
-			edits := myers.ComputeEdits(span.URIFromPath(buildSideString), buildSideString, probeSideString)
+			diffs, diffErr := generateDiff(buildSideFilePath, probeSideFilePath)
+
+			if diffErr != nil {
+				errorFileCount++
+				continue
+			}
+
 			fileCompared++
-			if len(edits) > 0 {
+			if len(diffs) > 0 {
 				diffFilePath := filepath.Join(OutputPath, fileId+".diff")
 				diffFile, ioErr := os.OpenFile(diffFilePath, utils.OpenNewFileFlags, 0644)
```

Comment:
Now you are already using `os.WriteFile` below, you do not need to open the file again here..